src/instructlab/data/generate.py

```diff
@@ -195,6 +207,16 @@ def generate(
         except Exception as exc:
             click.secho(f"Failed to start server: {exc}", fg="red")
             raise click.exceptions.Exit(1)
+
+        # If batch size is not set explicitly, disable batching when running
+        # with the local llama.cpp server
```

Comment:
this isn't detecting llama_cpp vs vllm, AIUI the local backend_instance could be either and we only have a (known) issue using concurrency with llama_cpp.

Reply:
Ah! Thanks for clarifying. I'll refine it further.

Reply:
It now does an `isinstance` check on the backend type after `select_backend`. Thanks for pointing this out.